src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,112 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { pattern } from '../../services/validation';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isImgUrlError, setIsImgUrlError] = useState(false);
+  const [isImdbUrlError, setIsImdbUrlError] = useState(false);
+
+  const handlerReset = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const handlerSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newMovie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    onAdd(newMovie);
+
+    setCount(currentCount => {
+      const newCount = currentCount + 1;
+
+      if (currentCount !== newCount) {
+        handlerReset();
+      }
+
+      return newCount;
+    });
```

Comment:
better not to use other functions (`handlerReset`) in state callback
```suggestion
    handlerReset();
    setCount(currentCount => currentCount + 1);
```